pkg/lib/accountmanagement/token.go

```diff
@@ -9,22 +9,41 @@ import (
 )
 
 type Token struct {
-	AppID       string     `json:"app_id,omitempty"`
-	UserID      string     `json:"user_id,omitempty"`
-	Alias       string     `json:"alias,omitempty"`
-	RedirectURI string     `json:"redirect_uri,omitempty"`
-	State       string     `json:"state,omitempty"`
-	TokenHash   string     `json:"token_hash,omitempty"`
-	CreatedAt   *time.Time `json:"created_at,omitempty"`
-	ExpireAt    *time.Time `json:"expire_at,omitempty"`
+	AppID     string     `json:"app_id,omitempty"`
+	UserID    string     `json:"user_id,omitempty"`
+	TokenHash string     `json:"token_hash,omitempty"`
+	CreatedAt *time.Time `json:"created_at,omitempty"`
+	ExpireAt  *time.Time `json:"expire_at,omitempty"`
+
+	// Adding OAuth
+	Alias       string `json:"alias,omitempty"`
+	RedirectURI string `json:"redirect_uri,omitempty"`
+	State       string `json:"state,omitempty"`
+
+	// Adding Phone
+	PhoneNumber string `json:"phone_number,omitempty"`
+
+	// Adding Email
+	Email string `json:"email,omitempty"`
+
+	// Updating Identity
+	IdentityID string `json:"identity_id,omitempty"`
 }
 
-func (t *Token) CheckUser(userID string) error {
+func (t *Token) CheckUserWithError(userID string, err error) error {
 	if subtle.ConstantTimeCompare([]byte(t.UserID), []byte(userID)) == 1 {
 		return nil
 	}
 
-	return ErrOAuthTokenNotBoundToUser
+	return err
+}
+
+func (t *Token) CheckOAuthUser(userID string) error {
+	return t.CheckUserWithError(userID, ErrOAuthTokenNotBoundToUser)
+}
+
+func (t *Token) CheckUser(userID string) error {
+	return t.CheckUserWithError(userID, ErrAccountManagementTokenNotBoundToUser)
```

Comment:
```go
func (t Token) CheckUser(userID string) error {
  return ErrAccountManagementTokenNotBoundToUser
}
```
and then at the caller site, if it is the oauth related operations, throw oauth specific error